src/sentry/api/endpoints/organization_events.py

```diff
@@ -94,28 +93,65 @@
 
 API_TOKEN_REFERRER = Referrer.API_AUTH_TOKEN_EVENTS.value
 
-DEFAULT_RATE_LIMIT = 30
-DEFAULT_RATE_LIMIT_WINDOW = 1
-DEFAULT_CONCURRENT_RATE_LIMIT = 15
-
-DEFAULT_EVENTS_RATE_LIMIT_CONFIG = {
-    "GET": {
-        RateLimitCategory.IP: RateLimit(
-            DEFAULT_RATE_LIMIT, DEFAULT_RATE_LIMIT_WINDOW, DEFAULT_CONCURRENT_RATE_LIMIT
-        ),
-        RateLimitCategory.USER: RateLimit(
-            DEFAULT_RATE_LIMIT, DEFAULT_RATE_LIMIT_WINDOW, DEFAULT_CONCURRENT_RATE_LIMIT
-        ),
-        RateLimitCategory.ORGANIZATION: RateLimit(
-            DEFAULT_RATE_LIMIT, DEFAULT_RATE_LIMIT_WINDOW, DEFAULT_CONCURRENT_RATE_LIMIT
-        ),
-    }
-}
-
+LEGACY_RATE_LIMIT = dict(limit=30, window=1, concurrent_limit=15)
+# reduced limit will be the future default for all organizations not explicitly on increased limit
+DEFAULT_REDUCED_RATE_LIMIT = dict(
+    limit=1000, window=300, concurrent_limit=15  # 1000 requests per 5 minutes
+)
+DEFAULT_INCREASED_RATE_LIMIT = dict(limit=50, window=1, concurrent_limit=50)
+
+
+def rate_limit_events(
+    request: Request, organization_slug: str | None = None, *args, **kwargs
+) -> dict[str, dict[RateLimitCategory, RateLimit]]:
+    """
+    Decision tree for rate limiting for organization events endpoint.
+    ```mermaid
+     flowchart TD
+         A[Get organization] --> B{Organization\nexists}
+         B -->|No| C[Return legacy rate limit]
+         B -->|Yes| D{Organization\nin increased\nrate limit}
+         D -->|Yes| E[Return increased rate limit]
+         D -->|No| F{Organization in\nreduced limit\nroll-out}
+         F -->|Yes| G[Return reduced rate limit]
+         F -->|No| H[Return legacy rate limit]
+     ```
+    """
+
+    def _config_for_limit(limit: RateLimit) -> dict[str, dict[RateLimitCategory, RateLimit]]:
+        return {
+            "GET": {
+                RateLimitCategory.IP: limit,
+                RateLimitCategory.USER: limit,
+                RateLimitCategory.ORGANIZATION: limit,
+            }
+        }
+
+    rate_limit = RateLimit(**LEGACY_RATE_LIMIT)
+
+    try:
+        organization = Organization.objects.get_from_cache(slug=organization_slug)
+    except Organization.DoesNotExist:
+        logger.warning("organization.slug.invalid", extra={"organization_slug": organization_slug})
+        return _config_for_limit(rate_limit)
+
+    if organization.id in options.get("api.organization_events.rate-limit-increased.orgs", []):
+        rate_limit = RateLimit(
+            **options.get(
```

Comment:
is it worth handling the case where someone incorrectly configures the option with invalid values? just thinking ahead for future people setting these -- how could that go wrong, if at all?

Reply:
good point, I'll add handling that and falling back to defaults.

Reply:
maybe in the future we can have schema validation for options :D

Reply:
I've added validation helper.